llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -1887,6 +1887,17 @@ OpenMPIRBuilder::createTask(const LocationDescription &Loc,
       SplitBlockAndInsertIfThenElse(IfCondition, IfTerminator, &ThenTI,
                                     &ElseTI);
       Builder.SetInsertPoint(ElseTI);
+
+      if (Dependencies.size()) {
+        Function *TaskWaitFn =
+            getOrCreateRuntimeFunctionPtr(OMPRTL___kmpc_omp_taskwait_deps_51);
```

Comment:
Since the `nowait` argument passed is false, wouldn't `__kmpc_omp_wait_deps` be sufficient?

Reply:
Will do.